src/goto-programs/remove_function_pointers.cpp

```diff
@@ -535,10 +601,57 @@ void remove_function_pointers(message_handlert &_message_handler,
   bool add_safety_assertion,
   bool only_remove_const_fps)
 {
+
   remove_function_pointers(
     _message_handler,
     goto_model.symbol_table,
     goto_model.goto_functions,
     add_safety_assertion,
     only_remove_const_fps);
 }
+
+void remove_function_pointerst::get_all_writes_to_pointers(
+  goto_functionst &goto_functions)
+{
+  for(goto_functionst::function_mapt::iterator f_it =
+        goto_functions.function_map.begin();
+      f_it != goto_functions.function_map.end();
+      f_it++)
+  {
+    goto_programt &goto_program = f_it->second.body;
+    Forall_goto_program_instructions(target, goto_program)
+    {
+      if(!target->is_assign())
+        continue;
+
+      const code_assignt &code_assign = to_code_assign(target->code);
+      const auto &lhs = code_assign.lhs();
+      const auto &rhs = code_assign.rhs();
+
+      if(lhs.type().id() == ID_pointer && lhs.type().subtype().id() == ID_code)
+      {
+        irep_idt lhs_irep;
+        if(lhs.id() == ID_symbol)
+          lhs_irep = to_symbol_expr(lhs).get_identifier();
+        else if(lhs.id() == ID_member)
+        {
+          lhs_irep = get_member_identifier(to_member_expr(lhs));
+        }
+        else
+          continue;
+
+        if(rhs.id() == ID_address_of && rhs.type().subtype().id() == ID_code)
+        {
+          irep_idt rhs_irep =
+            to_symbol_expr(to_address_of_expr(rhs).object()).get_identifier();
+          if(function_pointer_map.find(lhs_irep) != function_pointer_map.end())
+            (function_pointer_map[lhs_irep]).insert(rhs_irep);
```

Comment:
This if block currently redundant (`function_pointer_map[lhs_irep].insert(rhs_irep)` does both cases, no need for the ` = {{rhs_rep}} ` special case)